hazelcast-jet-s3/src/main/java/com/hazelcast/jet/s3/S3Sinks.java

```diff
@@ -0,0 +1,151 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.hazelcast.jet.function.FunctionEx;
+import com.hazelcast.jet.function.SupplierEx;
+import com.hazelcast.jet.pipeline.Sink;
+import com.hazelcast.jet.pipeline.SinkBuilder;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Contains factory methods for creating AWS S3 sinks.
+ */
+public final class S3Sinks {
+
+    private static final int DEFAULT_LINES_PER_FILE = 10240;
+
+    private S3Sinks() {
+    }
+
+    /**
+     * Convenience for {@link #s3(String, int, SupplierEx, FunctionEx)}.
+     * Creates an S3 client with given parameters. Uses {@link
+     * Object#toString()} for {@code toStringFn}.
+     */
+    @Nonnull
+    public static <T> Sink<? super T> s3(
+            @Nonnull String bucketName,
+            @Nonnull S3Parameters parameters
+    ) {
+        return s3(bucketName, DEFAULT_LINES_PER_FILE, parameters, Object::toString);
+    }
+
+    /**
+     * Convenience for {@link #s3(String, int, SupplierEx, FunctionEx)}.
+     * Creates an S3 client with given parameters.
+     */
+    @Nonnull
+    public static <T> Sink<? super T> s3(
+            @Nonnull String bucketName,
+            int linesPerFile,
+            @Nonnull S3Parameters parameters,
+            @Nonnull FunctionEx<? super T, String> toStringFn
+    ) {
+        return s3(bucketName, linesPerFile, () -> S3Utils.client(parameters), toStringFn);
+    }
+
+    /**
+     * Creates an AWS S3 {@link Sink} which writes items to files into the
+     * given bucket. Sink converts each item to string using given {@code
+     * toStringFn} and writes it as a line. Each processor will create a file
+     * for a batch of items. Name of the file will be processor's global index
+     * followed by an incremented counter, for example the processor having the
+     * index 2 will create these consecutive files {@code 2-0, 2-1, 2-2 ...}.
+     * <p>
+     * No state is saved to snapshot for this sink. If the job is restarted
+     * previously written files will be overwritten.
+     * <p>
+     * The default local parallelism for this sink is 1.
+     *
+     * @param bucketName     the name of the bucket
+     * @param linesPerFile   the number of lines per file
+     * @param clientSupplier S3 client supplier
+     * @param toStringFn     the function which converts each item to its
+     *                       string representation
+     * @param <T>            type of the items the sink accepts
+     */
+    @Nonnull
+    public static <T> Sink<? super T> s3(
+            @Nonnull String bucketName,
+            int linesPerFile,
+            @Nonnull SupplierEx<? extends AmazonS3> clientSupplier,
```

Comment:
I was looking at the S3 API and there's already a good builder for S3 client in the API. I don't think we need any other overload and this one. We can remove the S3Utils/S3Parameters and with examples in the Javadoc. This way we don't need so many overloads.